frontend/src/entities/jobs/algorithms.tsx

```diff
@@ -169,284 +218,343 @@ export const ML_ALGORITHMS: Algorithm[] = [
-            zValidator: integerIntervalValidator([1, Number.POSITIVE_INFINITY], IntervalStrategy.RIGHT_OPEN),
+            zValidator: numberValidator('learning_rate',
+                {   
+                    min: 1,
+                    required: true
+                })
         }, {
-            type: HyperparameterType.STATIC,
-            typeOptions: [HyperparameterType.STATIC],
+            ...staticParameterProperties,
             key: 'booster',
             value: ['gbtree'],
             description: 'Which booster to use. The gbtree and dart values use a tree-based model, while gblinear uses a linear function.',
             options: ['gbtree', 'gblinear', 'dart']
         }, {
-            type: HyperparameterType.STATIC,
-            typeOptions: [HyperparameterType.STATIC],
+            ...staticParameterProperties,
             key: 'verbosity',
             value: ['1'],
             options: [0, 1, 2, 3],
             description: 'Verbosity of printing messages.',
         }, {
-            type: HyperparameterType.STATIC,
-            typeOptions: [HyperparameterType.STATIC],
+            ...staticParameterProperties,
             key: 'nthread',
             value: [],
             description: 'Number of parallel threads used to run xgboost.',
-            zValidator: integerIntervalValidator([1, Number.POSITIVE_INFINITY], IntervalStrategy.RIGHT_OPEN).optional()
+            zValidator: numberValidator('learning_rate',
+                {   
+                    min: 1
+                })
         }, {
-            type: HyperparameterType.CONTINUOUS,
-            typeOptions: [HyperparameterType.CONTINUOUS, HyperparameterType.STATIC],
+            ...continuousParameterProperties,
             scalingType: 'Linear',
             key: 'eta',
             value: ['0.3'],
             description: 'Step size shrinkage used in updates to prevent overfitting. After each boosting step, you can directly get the weights of new features. The eta parameter actually shrinks the feature weights to make the boosting process more conservative.',
-            zValidator: floatIntervalValidator([0,1], IntervalStrategy.CLOSED).optional()
+            zValidator: numberValidator('eta',
+                {   
+                    min: 0, 
+                    max: 1,
+                    isFloat: true 
+                })
         }, {
-            type: HyperparameterType.CONTINUOUS,
-            typeOptions: [HyperparameterType.CONTINUOUS, HyperparameterType.STATIC],
+            ...continuousParameterProperties,
             scalingType: 'Linear',
             key: 'gamma',
             value: ['0'],
             description: 'Minimum loss reduction required to make a further partition on a leaf node of the tree. The larger, the more conservative the algorithm is.',
-            zValidator: floatIntervalValidator([0,Number.POSITIVE_INFINITY], IntervalStrategy.RIGHT_OPEN).optional()
+            zValidator: numberValidator('gamma',
```

Comment:
I think all these number validators can be all in the same line. Weird that our linters did not change that. Is pre-commit installed in your dev environment? 

We would probably not want to rely on pre-commit and our github actions should be catching this.

Reply:
I was doing it in the dev container. I am used to the `npm run start` catching these instantly which did not occur. I will try a build action.

Reply:
Linter still doesn't seem to be working. Moving on without it for now